bitcoin/src/lib.rs

```diff
@@ -885,19 +892,29 @@ impl BitcoinCoreApi for BitcoinCore {
         .await?;
 
         let proof = retry(get_exponential_backoff(), || async {
-            Ok(self.get_proof(txid, &block_hash).await?)
-        })
-        .await?;
-
-        let raw_tx = retry(get_exponential_backoff(), || async {
-            Ok(self.get_raw_tx(&txid, &block_hash).await?)
+            // fetch coinbase info..
+            let block = self.get_block(&block_hash).await?;
+            let coinbase_tx = block.coinbase().ok_or(Error::CoinbaseFetchingFailure)?;
+            let coinbase_txid = coinbase_tx.txid();
+            let coinbase_tx_proof = self.get_proof(coinbase_txid, &block_hash).await?;
+            let raw_coinbase_tx = self.get_raw_tx(&coinbase_txid, &block_hash).await?;
+
+            // fetch user tx info..
+            let raw_user_tx = self.get_raw_tx(&txid, &block_hash).await?;
+            let user_tx_proof = self.get_proof(txid, &block_hash).await?;
```

Comment:
We could join these but since the underlying connection is anyway synchronous I guess it won't make a difference